src/pages/rental/rental.ts

```diff
@@ -33,7 +33,17 @@ export class RentalPage {
 
   onAdd() {
     this.inventoryData.getItem(this.tag).subscribe(
-      item => this.items.push(item),
+      item => {
+        if (item.available === 0 && this.action === Actions.rent) {
+          this.stockpileData.showToast(Messages.itemAlreadyRented);
+        } else if (item.available === 1 && this.action === Actions.return) {
+          this.stockpileData.showToast(Messages.itemNotRented);
+        } else if (this.items.find(listItem => listItem.tag === item.tag)) {
```

Comment:
I think it would be more appropriate to use `array.prototype.some()` here, because it returns a boolean.
```JavaScript
} else if (this.items.some(listItem => listItem.tag === item.tag)) {`
```